packages/runtime/src/next/api-handler/handlers/webhook/site-published.ts

```diff
@@ -0,0 +1,10 @@
+import { revalidateTag } from 'next/cache'
+import { SitePublishedWebhookPayload, WebhookHandlerResult } from './types'
+
+export function handleSitePublished(_payload: SitePublishedWebhookPayload): WebhookHandlerResult {
+  revalidateTag(MAKESWIFT_CACHE_TAG)
+
+  return { body: { success: true }, status: 200 }
+}
+
+export const MAKESWIFT_CACHE_TAG = '@@makeswift'
```

Comment:
Should this be site specific?

For the multi-tenant case

Reply:
That's a good question. I didn't think about multi-tenant cases.

We could make it site-specific by using the site-api key. But I don't think it'll solve the multi-tenancy by itself. Perhaps this is also a reason why we need to expose `onPublish` option?

We also need to discuss what's the args for that `onPublish`. Pass the diff to `onPublish`?

For our default host, even right now we have a custom API handler, so I don't think we need to worry about our default host for now.

Reply:
Why do we need `onPublish` for this? We should be able to retrieve the site and use its ID as part of the tag

Reply:
> We could get the siteId by fetching by siteApiKey. And then, instead of using @@makeswift as the global tag, we use siteId as the tag. When a site is published, we call revalidateTag(siteId).

Yes, that is exactly what I mean.

I'm not sure I see this as an intermediate step only. We might still want to keep support for being able to revalidate everything in an entire Makeswift site (e.g., having a button in the builder to revalidate all). Having a hardcoded string isn't quite there yet given it doesn't represent a Makeswift site, but a site ID does. Multi-tenant hosts will be able to get the site ID given the `Host` header without any issue

Reply:
This is doable, but we decided to not do this for now.